package myvi.proto;

option java_package = "my.app.myvi.protobuf.gen";
option java_outer_classname = "MyviProtos";
option optimize_for = LITE_RUNTIME;

message charmap_pair_t {
	required uint32 glyph_index = 1;
	required uint32 ucode = 2;
}

message glyph_t {
	required uint32 glyph_index = 1;
	required sint32 font_height = 2;
	
	required sint32 bw = 3;
	required sint32 bh = 4;
	required sint32 bitmap_top = 6;
	required sint32 bitmap_left = 7;
	required sint32 mw = 8;
	required sint32 mh = 9;
	required sint32 hadvance = 10;
	required sint32 vadvance = 11;
	required uint32 buf_sz = 12;
	
	required bytes data = 13;
	optional uint32 buf_tag = 14;
}

message font_handle_t {
	repeated glyph_t root = 1;
	repeated charmap_pair_t charmap = 2;	
	required bytes name = 3;
	required sint32 curr_height = 4;
	optional uint32 first_charmap_tag = 5;
	optional uint32 cur_charmap_tag = 6;
	optional uint32 first_root_tag = 7;
	optional uint32 cur_root_tag = 8;
}

message ttcache_t {
	repeated font_handle_t handles = 1;
}
